### ISSUE TITLE
IEnforceRules types does not exist when use enforceExtended in TypeScript

### ISSUE THREAD

**Issue by dachi023** [ealush/vest#374]
I'm not very good at English, so I'm sorry if my sentences aren't right.

I get the following error when I try to use from TypeScript:

```ts
import vest, { test } from 'vest'
import enforce from 'vest/enforceExtended'

const validate = vest.create('form', (data: { message: string }) => {
  test('message', 'message is Required', () => {
    enforce(data.message).isNotEmpty()
  })
})

export default validate
```

```bash
# yarn run tsc
src/validate.ts:6:27 - error TS2339: Property 'isNotEmpty' does not exist on type 'EnforceExtendMap<ExtendedRules>'.

6     enforce(data.message).isNotEmpty()
                            ~~~~~~~~~~
```

This is just a suggestion, but we'll fix [vest.d.ts](https://github.com/ealush/vest/blob/2cc0d25b9dcb94ce149c63fe7498904cb39afe33/packages/vest/src/vest.d.ts#L100-L102):

```ts
type EnforceExtendMap<T> = IEnforceRules<T> & {
  [K in keyof T]: (...args: any[]) => EnforceExtendMap<T>;
};
```

And replace this code:

`IEnforceRules<T> & EnforceExtendMap<T>` → `EnforceExtendMap<T>`

If it's OK, I'll create a pull-request.

**Comment by ealush** [comment]
Hey @dachi023, thank you for reaching out. Sorry that you're experiencing this typing issue.

Saw your suggestion, I think this could work. Sure, go ahead and open a PR. We'll test that it works correctly.

Thank you!

**Comment by ealush** [comment]
@dachi023 Hey, I went ahead and fixed it myself, because I found a few more things that could be improved.

This will be added to the next minor version of vest in the next couple of days. Thank you!

**Comment by dachi023** [comment]
@ealush Waiting for the release! Thank you!

**Comment by ealush** [comment]
It's out now, vest 2.2.0. Let me know if that works for you.

**Comment by dachi023** [comment]
@ealush 

Upgrading to 2.2.0 is still giving me the error.

```ts
declare function enforceExtended(value: any): EnforceExtendMap<ExtendedRules>;
```

The type definition of the `EnforceExtendMap<ExtendedRules>` does not include `IEnforceRules`.

So, if you try to execute a method of `IEnforceRules` right after `enforce()`, the property does not exist.

```ts
// It works because the return value is of type `IEnforceRules<T> & EnforceExtendMap<T>`.
enforce(value).isURL().equals('https://github.com/ealush/vest')

// error: Property 'isNotEmpty' does not exist on type 'EnforceExtendMap<ExtendedRules>'.
// The `EnforceExtendMap<T>` contains only the properties of `ExtendedRules`, which causes an error.
enforce(value).equals('https://github.com/ealush/vest')
```

**Comment by ealush** [comment]
Oh, I see the problem. You are right.

Can you confirm that `vest@2.2.1-dev-b35d2f` works on your end? If so, I will publish it.

**Comment by dachi023** [comment]
I've upgraded and confirmed that the error no longer appears.

Thanks for fixing it!

**Comment by ealush** [comment]
Awesome, I've published 2.2.1 with this fix!
Thank you for the quick feedback loop.
